testing/upload_validation.ts

```diff
@@ -480,12 +475,83 @@ const booleanPropertySchema = zodCompleteObject<BooleanSchema & ObjectSchemaProp
   ...basePropertyValidators,
 });
 
-const numberPropertySchema = zodCompleteObject<NumberSchema & ObjectSchemaProperty>({
+const baseNumberPropertySchema = zodCompleteStrictObject<BaseNumberSchema & ObjectSchemaProperty>({
+  type: zodDiscriminant(ValueType.Number),
+  ...basePropertyValidators,
+})
+
+const numericPropertySchema = baseNumberPropertySchema.extend({
+  codaType: zodDiscriminant(ValueHintType.Percent).optional(),
+  precision: z.number().optional(),
+  useThousandsSeparator: z.boolean().optional(),
+})
+
+const scalePropertySchema = baseNumberPropertySchema.extend({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.Scale),
+  maximum: z.number().optional(),
+  icon: z.nativeEnum(ScaleIconSet).optional(),
+  ...basePropertyValidators,
+})
+
+const sliderPropertySchema = baseNumberPropertySchema.extend({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.Slider),
+  maximum: z.number().optional(),
+  minimum: z.number().optional(),
+  step: z.number().optional(),
+  ...basePropertyValidators,
+})
+
+const currencyPropertySchema = baseNumberPropertySchema.extend({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.Currency),
+  precision: z.number().optional(),
+  currencyCode: z.string().optional(),
+  format: z.nativeEnum(CurrencyFormat).optional(),
+  ...basePropertyValidators,
+})
+
+const numericDatePropertySchema = zodCompleteObject<NumericDateSchema & ObjectSchemaProperty>({
```

Comment:
We need similar zod schemas for the 3 string-based date/time schemas too, right?

Reply:
I was going to tackle them in their own followup PR, but you're right that these should probably be done in the same change 😅